app/screens/clock_display/clock_display_base.js

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {intlShape} from 'react-intl';
+
+import Preferences from 'mattermost-redux/constants/preferences';
+
+export default class ClockDisplayBase extends PureComponent {
+    static propTypes = {
+        theme: PropTypes.object.isRequired,
+        militaryTime: PropTypes.string.isRequired,
+        userId: PropTypes.string.isRequired,
+        actions: PropTypes.shape({
+            savePreferences: PropTypes.func.isRequired
+        }).isRequired
+    };
+    static contextTypes = {
+        intl: intlShape.isRequired
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            militaryTime: props.militaryTime
+        };
+    }
+
```

Comment:
I think you'll need to add a `componentWillReceiveProps` method to be able to actually control the prop change for Android and also it may be that while the setting is changed in another device while this screen is opened or something. but I'm not that sure.. thoughts?

Reply:
I'm not following.
this.props.militaryTime is used to reset the state as explained above. Where we are setting state to the props value